apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,353 @@
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+
+import { nodeModulesAssetPath } from '../../lib/assets';
+
+export const AdministratorQueryResultSchema = z.object({
+  rows: z.array(z.record(z.any())),
+  rowCount: z.number().optional(),
+  columns: z.array(z.string()).optional(),
+});
+export type AdministratorQueryResult = z.infer<typeof AdministratorQueryResultSchema>;
+
+export const AdministratorQueryRunParamsSchema = z.object({
+  name: z.string(),
+  sql: z.string(),
+  params: z.record(z.any()),
+  authn_user_id: z.string(),
+  error: z.string().optional().nullable(),
+  result: AdministratorQueryResultSchema.optional().nullable(),
+  formatted_date: z.string().optional().nullable(),
+});
+export type AdministratorQueryRunParams = z.infer<typeof AdministratorQueryRunParamsSchema>;
+
+const AdministratoryQueryParamsRecordSchema = z.record(z.string());
+export type AdministratorQueryParamsRecord = z.infer<typeof AdministratoryQueryParamsRecordSchema>;
```

Comment:
AFAICT, this schema is never actually used, and the type is trivial. I'd recommend just inlining `Record<string, string>` in the one place this type is used. In general: if a schema isn't actually used, just write the type directly.

Reply:
Removed this schema and wrote the type directly in commit 53b8ebc.